lib/CFRPageActions.jsm

```diff
@@ -275,6 +284,47 @@ class PageAction {
     }
   }
 
+  _createElement({type, id}, parent) {
+    let element = this.window.document.createElement(type);
+    if (id) {
+      element.setAttribute("id", id);
+    }
+    parent.appendChild(element);
+    return element;
+  }
+
+  async _renderPinTabAnimation() {
+    const ANIMATION_CONTAINER_ID = "cfr-notification-footer-pintab-animation-container";
+    const footer = this.window.document.getElementById("cfr-notification-footer");
+    let animationContainer = this.window.document.getElementById(ANIMATION_CONTAINER_ID);
+    if (!animationContainer) {
+      animationContainer = this._createElement({type: "vbox", id: ANIMATION_CONTAINER_ID}, footer);
+
+      // spacer
+      this._createElement("vbox", animationContainer).setAttribute("flex", 1);
+
+      let controlsContainer = this._createElement({type: "hbox", id: "cfr-notification-footer-animation-controls"}, animationContainer);
+
+      // spacer
+      this._createElement({type: "vbox"}, controlsContainer).setAttribute("flex", 1);
+
+      let pauseButton = this._createElement({type: "hbox", id: PAUSE_BUTTON_ID}, controlsContainer);
+
+      let pauseLabel = this._createElement({type: "label", id: "cfr-notification-footer-pause-label"}, pauseButton);
+      pauseLabel.textContent = await this.getStrings({"string_id": "cfr-doorhanger-pintab-animation-pause"});
+
+      // pause icon
+      this._createElement({type: "image", id: "cfr-notification-footer-pause-icon"}, pauseButton);
+    }
+
+    animationContainer.toggleAttribute("animate", Services.prefs.getBoolPref(ANIMATIONS_ENABLED_PREF, true));
+    animationContainer.removeAttribute("paused");
+
+    let pauseButton = this.window.document.getElementById(PAUSE_BUTTON_ID);
+    this.onPauseClick = () => { animationContainer.setAttribute("paused", true); };
```

Comment:
you could also put these lines under `if !(this.onPauseClick) { }` just in case

Reply:
yep, that's a good idea 👍